### ISSUE TITLE
Add `blocks.append(type, data)` API method

### ISSUE THREAD

**Issue by gohabereg** [codex-team/editor.js#715]
This method should allow to append (or insert) new Blocks programmatically

Example:

```
editor.blocks.append('paragraph', {text: 'Hello world'});

editor.blocks.insert(4, 'header', {text: 'Hello world'});
```

**Comment by neSpecc** [comment]
I thought, the position can be an optional argument of the `append`

**Comment by gohabereg** [comment]
Better to name it `insert` then

**Comment by khaydarov** [comment]
I tend to think that it is better to use output format as a argument of append function, like 
```js
{
  type: 'header',
  data: { 
    ....
  } 
}
```

and second argument will be `position`

```
const newBlock = {
  type: 'paragraph', 
  data: {
     text: 'Hello world'
  }
};
editor.blocks.append(newBlock, 4);
```

**Comment by neSpecc** [comment]
@cjagannath please, describe the use case of that feature. How you would generate a block's data structure? It is an internal logic of a plugin. And `type` is the variable that defined by plugins users.

I can't see a good implementation of it, except hardcoding.

**Comment by dimensi** [comment]
@neSpecc for inline tool for example. We can use this api for extract text from selection and create new Header or Paragraph element.

**Comment by neSpecc** [comment]
@dimensi to create a Header, you should pass for example

```
{
   text: 'Extracted fragment',
   level: 3
}
```

How you will create this structure? Your Inline Tool doesn't know about properties that used by Header data (I mean `text` and `label`). So you are only able to hard code them.

**Comment by neSpecc** [comment]
For blocks convertation we already develop an ability (#704). Any other use cases?

**Comment by dimensi** [comment]
> @dimensi to create a Header, you should pass for example
> 
> ```
> {
>    text: 'Extracted fragment',
>    level: 3
> }
> ```
> How you will create this structure? Your Inline Tool doesn't know about properties that used by Header data (I mean `text` and `label`). So you are only able to hard code them.

Yes hardcode, for you use inline tools like this, inline tool must check through validation whether or not there's a Header class. Something like that
```js
{
  render() {
   const blocks = this.api.getBlocks()  // this api not implement
   const hasHeader = blocks.findIndex(el => el.class instanceof Header)
   if (hasHeader) return button;
   return;
  }
}
```

**Comment by cjagannath** [comment]
> @cjagannath please, describe the use case of that feature. How you would generate a block's data structure? It is an internal logic of a plugin. And `type` is the variable that defined by plugins users.
> 
> I can't see a good implementation of it, except hardcoding.

For my use case, I would like to offer a custom tools experience where all the tools are listed on the left. Users can pick any tool and add it to the editor. The screenshot below would give an idea. The default tools menu(+) would have a sub set of the tools.

![image](https://user-images.githubusercontent.com/5174344/55949804-0ff0a600-5c71-11e9-9ff7-687538e5ea3f.png)

As a user, I would know the list of plugins that I want to use. Some plugins will be built by me for my specific case. So I know all their types and data structure. The API would make it very easy to pick and add tools.

If one has to think as an author, I agree it's hard coding the plug-in info. That's why in my request #717 I had asked about extending the tools menu UI. But that could become too complex.

Hope I am clear. Let me know if any more info is required.

Thank you for listening.

**Comment by gohabereg** [comment]
Resolved by #826
